### ISSUE TITLE
Umbraco.Tags stored as Csv don't get indexed in examine

### ISSUE THREAD

**Issue by MrJackWilson** [umbraco/Umbraco-CMS#15598]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

12.3.6

### Bug summary

Properties that use the Umbraco.Tags data type, where the data type's storage type is set to Csv, will not be indexed in examine. However, as soon as I change the storage type to Json, the properties appear.

### Steps to reproduce

- Create a fresh install of 12.3.6. (This problem also occurred on 11)
- Create a new property editor with the Umbraco.Tags data type and change the storage type to Csv
- Create a new document type and add a property using the new editor you just created
- Create a content node with that document type and add some tags, then save & publish
- Search examine for that node using __NodeTypeAlias
- The tags property you just created and populated will not appear in the list of fields
- Go back and change the storage type to Json
- Save & publish the node again
- Search examine and you should now see the tags field in there

### Expected result / actual result

No matter the storage type, the value of a tags property should appear in the list of examine fields.

**Comment by andr317c** [comment]
Hey! 
Thanks for taking the time to create this issue, we appreciate it! I was able to reproduce this on versions 12.3.6 and 13.0.3. 

Here is an example of what I did to reproduce this: 

I have created these two tag data types, one with CSV and one with JSON storage type in a published content node. 
![image](https://github.com/umbraco/Umbraco-CMS/assets/73799582/297d964a-4bfd-4841-941f-471c7fd6c49d)

When I then tried to search on the External/InternalIndex I only saw the Tags from the JSON storage type as mentioned in this issue 
![image](https://github.com/umbraco/Umbraco-CMS/assets/73799582/edfc83be-31d6-40ee-88cd-a82793e766c3)

I will go ahead and mark this issue as up for grabs 😄
